src/llv8-inl.h

```diff
@@ -83,6 +83,12 @@ inline int64_t HeapObject::GetType(Error& err) {
 inline int64_t Map::GetType(Error& err) {
   int64_t type = LoadField(v8()->map()->kInstanceAttrsOffset, err);
   if (err.Fail()) return -1;
+
+  if( v8()->process_.GetByteOrder() != HOST_BYTE_ORDER) {
+    // The type is only one byte within the instance attrs.
+    type = type >> 0x30;
```

Comment:
Since "because that works" wasn't a good answer I found the v8 comments explaining the way it works:
```
// Order instance type and bit field together such that they can be loaded
// together as a 16-bit word with instance type in the lower 8 bits regardless
// of endianess. Also provide endian-independent offset to that 16-bit word.
```
I've updated the code to read a 16 bit unsigned value and that works both ways without a special case.